keps/sig-api-machinery/20190325-unions.md

```diff
@@ -233,6 +233,41 @@ This process works as follows:
+| A11      | `discriminator: A`<br>`a: {...}` | `a: {...}`                       | `discriminator: A`<br>`a: {...}` |
+| A12      | `discriminator: A`<br>`a: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: A`<br>`a: {...}` |
+| A13      | `discriminator: B`<br>`b: {...}` | empty                            | `discriminator: B`<br>`b: {...}` |
+| A14      | `discriminator: B`<br>`b: {...}` | `a: {...}`                       | `discriminator: B`<br>`b: {...}` |
+| A15      | `discriminator: B`<br>`b: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>`b: {...}` |
+
+Example updates with invalid content in the new object:
+
+| Scenario | New object                       | Existing object                  | Result                                                                                                                                   |
+|----------|----------------------------------|----------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| B1       | `a: {...}, b: {...}`             | empty                            | `a: {...}, b: {...}`<br>(more than one field added, fail in validation with multiple set)                                                |
+| B2       | `a: {...}, b: {...}`             | `a: {...}`                       | `discriminator: B`<br>`b:{...}`<br>(one field added, drop other fields, pass validation)                                                 |
+| B3       | `a: {...}, b: {...}`             | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>`b:{...}`<br>(one field added, drop other fields, pass validation)                                                 |
+| B4       | `a: {...}, b: {...}`             | `discriminator: B`<br>`b: {...}` | `discriminator: A`<br>`a:{...}`<br>(one field added, drop other fields, pass validation)                                                 |
+| B5       | `discriminator: A`<br>`b: {...}` | empty                            | `discriminator: A`<br>`b:{...}`<br>(discriminator changed and different field added, fail in validation)                                 |
+| B6       | `discriminator: A`<br>`b: {...}` | `a: {...}`                       | `discriminator: A`<br>`b:{...}`<br>(discriminator changed and different field added, fail in validation)                                 |
+| B7       | `discriminator: A`<br>`b: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>`b:{...}`<br>(discriminator unchanged, one field added, drop other fields, autoset descriminator, pass validation) |
+| B8       | `discriminator: B`<br>`a: {...}` | empty                            | `discriminator: B`<br>`a:{...}`<br>(discriminator changed and different field added, fail in validation)                                 |
```

Comment:
The . fact that you write both the result AND the "fail in validation" is a little confusing :-)

Reply:
Because it will fail before going to validation (the normalizer will fail with an error and not return the object you described)